src/include/distributed/connection_cache.h

```diff
@@ -58,7 +58,7 @@ typedef enum
 	XACT_MODIFICATION_INVALID = 0, /* placeholder initial value */
 	XACT_MODIFICATION_NONE,        /* no modifications have taken place */
 	XACT_MODIFICATION_DATA,        /* data modifications (DML) have occurred */
-	XACT_MODIFICATION_SCHEMA       /* schema modifications (DDL) have occurred */
+	XACT_MODIFICATION_MULTI_SHARD  /* multi-shard modifications have occurred */
```

Comment:
I think this hides the real meaning of these values. In particular, it feels like we have three distinct failure modes:

  * Nothing needs taking care of (`SELECT`)
  * Mark shards as unhealthy (`INSERT`, classically)
  * Rollback 2PC (multi-shard stuff)

What do you think about moving in this direction (possibly not in this review)?